File: C_Programming/Practise_Programs/PP_185.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_185
//
//  Description       : This program toggles the 3rd and 8th bits of
//                      a given unsigned integer by combining two
//                      separate bit masks using the bitwise OR
//                      operator.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 23/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

typedef unsigned int UINT;

// Position 3rd and 8th
int main()
{
    UINT iMask1 = 0x00000004;
    UINT iMask2 = 0x00000080;
    UINT iMask = 0;
    
    UINT iNo = 0;
    UINT iResult = 0;

    printf("Enter Number : \n");
    scanf("%u", &iNo);

    iMask = iMask1 | iMask2;

    iResult = iNo ^ iMask;

    printf("Updated No : %u\n", iResult);

    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      10
//
//  Sample Output :
//
//      Enter Number :
//      Updated No : 142
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_185.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_185
//
//  Description       : This program swaps two numbers using a
//                      third variable.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 07/09/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class PP_185
{
    public static void Swap(int iNo1, int iNo2)
    {
        int temp = 0;

        System.out.println("Before Swap First Number is : " + iNo1);
        System.out.println("Before Swap Second Number is : " + iNo2);

        temp = iNo1;        
        iNo1 = iNo2;        
        iNo2 = temp;        

        System.out.println("After Swap First Number is : " + iNo1);
        System.out.println("After Swap Second Number is : " + iNo2);        
    }
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue1 = 0, iValue2 = 0, iRet = 0;

        System.out.println("Enter the First Number : ");
        iValue1 = sobj.nextInt();

        System.out.println("Enter the Second Number : ");
        iValue2 = sobj.nextInt();

        Swap(iValue1, iValue2);

        sobj.close();
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Enter the First Number :
//      10
//      Enter the Second Number :
//      20
//
//  Sample Output :
//      Before Swap First Number is : 10
//      Before Swap Second Number is : 20
//      After Swap First Number is : 20
//      After Swap Second Number is : 10
//
//////////////////////////////////////////////////////////////////